test/runtime/jhh/lps_test.py

```diff
@@ -284,22 +308,120 @@ def test_12_oversubscribed_no_env(self):
+                    '--ntasks=3',
                     '--ntasks-per-node=3', '--cpus-per-task=16',
                     '--exclusive', '--kill-on-bad-exit',
                     './lps_test_real', '-nl', '3', '-v'], env=env)
-        self.assertIn('oversubscribed = True', proc.stdout)
-        self.assertIn('Using 8 Shepherds', proc.stdout)
+        self.assertIn('oversubscribed = True', output)
+        self.assertIn('Using 8 Shepherds', output)
+
+# copied from sub_test.py
+# report an error message and exit
+def Fatal(message):
+    sys.stdout.write('[Error (sub_test): '+message+']\n')
+    magic_exit_code = reduce(operator.add, map(ord, 'CHAPEL')) % 256
+    sys.exit(magic_exit_code)
+
+def getDir(compiler):
+
+    # This code is copied from sub_test.py to ensure we print
+    # the path of the test in the same format so all paths look
+    # the same in the test outputs.
+
+    # Find the base installation
+    #compiler=argv[1]
+    if not os.access(compiler,os.R_OK|os.X_OK):
+        Fatal('Cannot execute compiler \''+compiler+'\'')
+
+    is_chpldoc = compiler.endswith('chpldoc')
+
+    path_to_compiler=os.path.abspath(os.path.dirname(compiler))
+    # Assume chpl binary is 2 directory levels down in the base installation
+    (chpl_base, tmp) = os.path.split(path_to_compiler)
+    (chpl_base, tmp) = os.path.split(chpl_base)
+    chpl_base=os.path.normpath(chpl_base)
+    # sys.stdout.write('CHPL_BASE='+chpl_base+'\n')
+
+    # If $CHPL_HOME is not set, use the base installation of the compiler
+
+    chpl_home=os.getenv('CHPL_HOME', chpl_base);
+    chpl_home=os.path.normpath(chpl_home)
+
+    # Find the test directory
+    testdir=chpl_home+'/test'
+    if os.path.isdir(testdir)==0:
+        testdir=chpl_home+'/examples'
+        if os.path.isdir(testdir)==0:
+            Fatal('Cannot find test directory '+chpl_home+'/test or '+testdir)
+    # Needed for MacOS mount points
+    testdir = os.path.realpath(testdir)
+    # sys.stdout.write('testdir='+testdir+'\n');
+
+    # If user specified a different test directory (e.g. with --test-root flag on
+    # start_test), use it instead.
+    test_root_dir = os.environ.get('CHPL_TEST_ROOT_DIR')
+    if test_root_dir is not None:
+        testdir = test_root_dir
+
+    # Get the current directory (normalize for MacOS case-sort-of-sensitivity)
+    localdir = os.path.normpath(os.getcwd()).replace(testdir, '.')
+    # sys.stdout.write('localdir=%s\n'%(localdir))
+
+    if localdir.find('./') == 0:
+        # strip off the leading './'
+        localdir = localdir.lstrip('.')
+        localdir = localdir.lstrip('/')
+    # sys.stdout.write('localdir=%s\n'%(localdir))
+
+    return localdir
 
 def main(argv):
     global verbose
     failfast = False
-    if "-f" in argv:
+    if "-f" in argv or "--failfast" in argv:
         failfast = True
-        argv.remove("-f")
-    if "-v" in argv:
+        try:
+            argv.remove("-f")
+            argv.remove("--failfast")
+        except:
```

Comment:
I think this could hide issues like syntax errors accidentally added in the `try`, can you be more specific on what error you catch (I think what you want is `ValueError`, but I could be wrong)